dali/pipeline/operators/reader/loader/file_loader.cc

```diff
@@ -107,7 +108,7 @@ void FileLoader::ReadSample(ImageLabelWrapper* image_label) {
     current_index_ = 0;
   }
 
-  FileStream *current_image = FileStream::Open(file_root_ + "/" + image_pair.first);
+  std::shared_ptr<FileStream> current_image(FileStream::Open(file_root_ + "/" + image_pair.first));
```

Comment:
Would `std::unique_ptr` not be more appropriate here?

Reply:
Agree with above. Also, if FileStream is our class, we could add a call to Close() in the destructor to make it more RAII-conformant

Reply:
@pribalta - not really, Open returns pointer to object of derived class which implements inside its own destructor call to close method - https://github.com/NVIDIA/DALI/blob/master/dali/util/local_file.h#L34. We cannot call close in the FileStream destructor itself as it won't call Close of the derived class.